### DISCUSSION TITLE
Static asset referred from CSS is not loading with file-loader webpack config override

### DISCUSSION THREAD

**Discussion by ayush000** [vercel/next.js#33094] [Help]
### What version of Next.js are you using?

12.0.7

### What version of Node.js are you using?

14.18.2

### What browser are you using?

Chrome

### What operating system are you using?

macOS Monterey 12.1

### How are you deploying your application?

I'm using NextJS Express integration

### Describe the Bug

I encountered this issue even during local development. When I override NextJS webpack config and add file-loader (v6.2.0), static assets referred in CSS files stop working. As soon as I remove file-loader, static assets in CSS files get loaded correctly.

### Expected Behavior

Adding file-loader to next config shouldn't break images referred from CSS files

### To Reproduce

For example: If contents of my `next.config.js` are:
```js
module.exports = {
  reactStrictMode: true,
  webpack: (config, { buildId, dev, isServer, defaultLoaders, webpack }) => {
    config.module.rules.push({
      test: /\.(jpe?g|png|gif|ico|webp|svg)$/,
      use: [
        {
          loader: 'file-loader',
        },
      ],
    });

    return config
  },
}
```
and my `global.css` file refers a static asset from `public/` directory:
```css
/* styles/global.css */
/* imported in pages/_app.js */

ul {
  list-style: url('../public/img.png');
}
```

then I get a broken image on my site. When I remove the file-loader override, it works fine. Also, JS image imports are working fine in both the cases.

I checked the `.next` folder. WIth file-loader, there are two png files generated for my image:
* `.next/static/[longhash].png`. This file is the actual image.
* `.next/static/media/[filename].[shorthash].png`. This file is corrupt as it doesn't contain the image. When I open it in a plain text editor, I found that it contains JS: `export default __webpack_public_path__ + "[longhash].png";`. A `png` file shouldn't contain any JS code

`.next/static/chunks/pages/_app.js` refers to `.next/static/media/[filename].[shorthash].png` which results in a corrupted image being loaded on the browser.

**Comment by ayush000** [comment]
@balazsorban44 I was wondering whether this should be a discussion or a bug. The import path work for JS files when you override file-loader, but doesn't work when you refer the exact same path in CSS files. Also, I couldn't really find a way to refer static files in CSS (with file-loader overrides). I didn't receive any answers for this discussion as well. So I feel it should be a bug if there is really no way to achieve this.

What do you think? Let me know if a reproduction in codesandbox / some other platform would be helpful.

**Reply by balazsorban44** [reply]
Since you are overriding the webpack config, I assumed there is a high possibility the error would originate from your config.

I'm not entirely sure why you would need it in the first place. You can reference files from the `public` folder like so, without any configuration:

```css
/* styles/global.css */
/* imported in pages/_app.js */

ul {
  list-style: url('/img.png');
}
```

**Reply by sami616** [reply]
I'm very much in need of being able to refer to an asset from my css and the above is also happening to me. Did you ever find an answer?
